### PR TITLE
Fixes exception on LoggerMessageAnalyzer for missing named placeholder

### PR DESCRIPTION
**Pull request by maryamariyan**
Fixes exception on LoggerMessageAnalyzer for missing named placeholder.

Fixes https://github.com/dotnet/roslyn-analyzers/issues/5362

### CODE REVIEW

**Comment by maryamariyan** [comment]
cc @pranavkm

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs [hunk 1]
```diff
@@ -223,7 +223,7 @@ private void AnalyzeFormatArgument(OperationAnalysisContext context, IOperation
                 {
                     context.ReportDiagnostic(formatExpression.CreateDiagnostic(CA2253Rule));
                 }
-                else if (char.IsLower(valueName[0]))
+                else if (valueName.Length > 0 && char.IsLower(valueName[0]))
```
Comment (pranavkm) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs:226]: Is the expected to be never null? Would we do `!string.IsNullOrEmpty` instead?

**Review by pranavkm — APPROVED** [review]

**Review by buyaa-n — APPROVED** [review]
LGTM

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs (hunk 1 above)
Comment (maryamariyan) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs:226]: sure we could switch over to that

**Review by jmarolf — APPROVED** [review]

**Pull request merged**